src/components/state-history-charts.ts

```diff
@@ -0,0 +1,118 @@
+import "./ha-circular-progress";
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import "./state-history-chart-line";
+import "./state-history-chart-timeline";
+import { isComponentLoaded } from "../common/config/is_component_loaded";
+import type { HomeAssistant } from "../types";
+import { HistoryResult } from "../data/history";
+
+@customElement("state-history-charts")
+class StateHistoryCharts extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property() public historyData!: HistoryResult;
+
+  @property() public names = false;
+
+  @property() public endTime?: Date;
+
+  @property() public upToNow = false;
+
+  @property() public noSingle = false;
+
+  @property() public isLoadingData = false;
+
+  protected render(): TemplateResult {
+    if (!isComponentLoaded(this.hass, "history")) {
+      return html` <div class="info">
+        ${this.hass.localize("ui.components.history_charts.history_disabled")}
+      </div>`;
+    }
+
+    if (this.isLoadingData && !this.historyData) {
+      return html` <div class="info">
+        ${this.hass.localize("ui.components.history_charts.loading_history")}
+      </div>`;
+    }
+
+    if (this._isHistoryEmpty()) {
+      return html` <div class="info">
+        ${this.hass.localize("ui.components.history_charts.no_history_found")}
+      </div>`;
+    }
+
+    return html`
+      ${this.historyData.timeline.length
+        ? html`
+            <state-history-chart-timeline
+              .hass=${this.hass}
+              .data=${this.historyData.timeline}
+              .endTime=${this._computeEndTime()}
+              .noSingle=${this.noSingle}
+              .names=${this.names}
+            ></state-history-chart-timeline>
+          `
+        : html``}
+      ${this.historyData.line.map(
+        (line) => html`
+          <state-history-chart-line
+            .hass=${this.hass}
+            .unit=${line.unit}
+            .data=${line.data}
+            .identifier=${line.identifier}
+            .isSingleDevice=${!this.noSingle &&
+            line.data &&
+            line.data.length === 1}
+            .endTime=${this._computeEndTime()}
```

Comment:
We should move this to a var above the `return html`, so we do it only once instead for every chart.